### ISSUE TITLE
optimize_hyperparameters on TFT finishes really early

### ISSUE THREAD

**Issue by YojoNick** [jdb78/pytorch-forecasting#707]
- PyTorch-Forecasting version:      0.9.0
- PyTorch version:                         1.9.0+cu111
- Python version:                           3.6.9
- Operating System:                      Ubutnu 18.04

### Expected behavior

When optimizing the hyperparameters for the Temporal Fusion Transformer, the function finishes after 9 trials even though I set the number of trials to 200.

### Actual behavior

I would have expected the function to continue to 200 trials.  

### Code to reproduce the problem

```
    study = optimize_hyperparameters(
        trainDataLoader,
        validationDataLoader,
        model_path='/path/to/model/',
        n_trials=200,
        max_epochs=50,
        gradient_clip_val_range=1e-2, 1e2,
        hidden_size_range=10,320,
        hidden_continuous_size_range=10, 320,
        attention_head_size_range=1,4,
        learning_rate_range=1e-4, 1e-2,
        dropout_range=0.1, 0.9,
        trainer_kwargs=30,
        reduce_on_plateau_patience=4,
        use_learning_rate_finder=False,  
    )
```

After execution, the code simply says 
> Trial 9 finished with value: 0.011759861372411251 and parameters: {'gradient_clip_val': 0.1525726587235903, 'hidden_size': 77, 'dropout': 0.27776972546206863, 'hidden_continuous_size': 51, 'attention_head_size': 1, 'learning_rate': 0.0021751372840664437}. Best is trial 2 with value: 0.0105184530839324.
> INFO:root:{'gradient_clip_val': 38.448708760246916, 'hidden_size': 156, 'dropout': 0.22089391108373357, 'hidden_continuous_size': 12, 'attention_head_size': 4, 'learning_rate': 0.001433637889287587}

Why did it stop early (9 trials instead of 200)? Is this ok/expected?
